src/context-selection/README.md

```diff
@@ -0,0 +1,48 @@
+# Context selection
```

Comment:
Having markdown files throughout the code seems a little unconventional to me. Doesn't have to be a bad thing of course. But it seems to me this readme is doing two things:

1. Describing the "contracts", i.e. "this module exports this, does that, etc."
2. Offer some explanation on how things work

I think I'd prefer 1. to be addressed with just unit tests. They should also describe the contracts, plus they will break if anything changes.

And for 2. I think I'd prefer comments in the code. If the code is complex enough that it's hard to comprehend then a comment closeby should help. Having it in the code also makes it easier to update it if the code changes.

For documentation like this I worry that it would get out of date faster than the above solutions would.

Reply:
Agreed, can be done entirely with tests actually